lib/token.js

```diff
@@ -469,48 +483,64 @@ function getWithPopup(sdk, oauthOptions, options) {
   return getToken(sdk, oauthParams, options);
 }
 
-function getWithRedirect(sdk, oauthOptions, options) {
-  oauthOptions = util.clone(oauthOptions) || {};
+function prepareOauthParams(sdk, oauthOptions) {
   var oauthParams = getDefaultOAuthParams(sdk, oauthOptions);
-  // If the user didn't specify a responseMode
-  if (!oauthOptions.responseMode) {
-    // And it's only an auth code request (responseType could be an array)
-    var respType = oauthParams.responseType;
-    if (respType.indexOf('code') !== -1 &&
-        (util.isString(respType) || (Array.isArray(respType) && respType.length === 1))) {
-        // Default the responseMode to query
-        util.extend(oauthParams, {
-          responseMode: 'query'
-        });
-    // Otherwise, default to fragment
-    } else {
-      util.extend(oauthParams, {
-        responseMode: 'fragment'
-      });
-    }
+
+  if (oauthParams.grantType !== 'authorization_code') {
+    return Q.resolve(oauthParams);
   }
 
-  var urls = oauthUtil.getOAuthUrls(sdk, oauthParams, options);
-  var requestUrl = urls.authorizeUrl + buildAuthorizeParams(oauthParams);
+  // PKCE authorization_code flow
+  var codeVerifier = sdk.pkce.generateVerifier(oauthParams.codeVerifier || '');
```

Comment:
No need for the `|| ''` here, as we're already setting it to an empty string when `generateVerifier ` is invoked.